include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -2033,16 +2034,65 @@ class graph_oriented_layout_design_impl
                                               nodes_to_place_breadth_co_to_ci, nodes_to_place_breadth_ci_to_co,
                                               nodes_to_place_depth_co_to_ci, nodes_to_place_depth_ci_to_co);
                 }
-            }
 
-            // Set cost objectives based on effort mode and cost objective
-            auto set_costs = [&](uint64_t start_idx, uint64_t end_idx, auto cost_objective)
-            {
-                for (uint64_t i = start_idx; i < end_idx; ++i)
+                if (ps.mode != graph_oriented_layout_design_params::effort_mode::HIGHEST_EFFORT)
                 {
-                    ssg_vec[i].cost = cost_objective;
+                    params.strategy = fanout_substitution_params::substitution_strategy::RANDOM;
+                    params.seed     = seed;
+                    mockturtle::fanout_view network_substituted_random{fanout_substitution<tec_nt>(ntk, params)};
+
+                    topo_view_co_to_ci_random<decltype(network_substituted_random)> network_breadth_co_to_ci_random{
+                        network_substituted_random, seed};
+                    topo_view_ci_to_co_random<decltype(network_substituted_random)> network_breadth_ci_to_co_random{
+                        network_substituted_random, seed};
+                    topo_view_co_to_ci_random<decltype(network_substituted_random)> network_depth_co_to_ci_random{
+                        network_substituted_random, seed};
+                    topo_view_ci_to_co_random<decltype(network_substituted_random)> network_depth_ci_to_co_random{
+                        network_substituted_random, seed};
+
+                    // prepare nodes to place for additional networks
+                    std::vector<mockturtle::node<decltype(network_breadth_co_to_ci_random)>>
+                        nodes_to_place_breadth_co_to_ci_random{};
+                    std::vector<mockturtle::node<decltype(network_breadth_ci_to_co_random)>>
+                        nodes_to_place_breadth_ci_to_co_random{};
+                    std::vector<mockturtle::node<decltype(network_depth_co_to_ci_random)>>
+                        nodes_to_place_depth_co_to_ci_random{};
+                    std::vector<mockturtle::node<decltype(network_depth_ci_to_co_random)>>
```

Comment:
As above: if you know their sizes, definitely reserve capacity ahead of usage to increase performance by avoiding multiple reallocations